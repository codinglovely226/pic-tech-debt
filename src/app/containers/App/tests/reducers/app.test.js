/* eslint-disable */

import casesReducer from 'containers/CaseEntry/reducers/cases';

import {
  batchInsertDocument,
  deleteDocument,
  displayRecord,
  insertDocument,
  updateDocument,
} from '../../actions';

describe('casesReducer', () => {
  var mockedDate = new Date(2018, 3, 7, 17, 8);
  global.Date = jest.fn(() => mockedDate);
  it('returns the initial state', () => {
    expect(casesReducer(undefined, {})).toMatchSnapshot();
  });
  it('batch insert case', () => {
    expect(casesReducer({}, batchInsertDocument([{'_id': 'Test Case 1'}, {'_id': 'Test Case 2'}]))).toMatchSnapshot();
  });
  it('delete case', () => {
    expect(casesReducer({'Test Case': {}}, deleteDocument('Test Case'))).toMatchSnapshot();
  });
  it('display record', () => {
    expect(casesReducer({'Test Case': {}}, displayRecord('Test Case'))).toMatchSnapshot();
  });
  it('insert case', () => {
    expect(casesReducer({}, insertDocument({'_id': 'Test Case'}))).toMatchSnapshot();
  });
  it('update case', () => {
    expect(casesReducer({'Test Case': {}}, updateDocument({'_id': 'Test Case', 'siteCreated': 2}))).toMatchSnapshot();
  });
});
