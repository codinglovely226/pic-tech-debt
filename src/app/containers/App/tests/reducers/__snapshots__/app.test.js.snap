exports[`casesReducer batch insert case 1`] = `Object {}`;

exports[`casesReducer delete case 1`] = `Object {}`;

exports[`casesReducer display record 1`] = `
Object {
  "Test Case": Object {},
}
`;

exports[`casesReducer insert case 1`] = `Object {}`;

exports[`casesReducer returns the initial state 1`] = `Object {}`;

exports[`casesReducer update case 1`] = `
Object {
  "Test Case": Object {},
}
`;
